### ISSUE TITLE
Can we hook something before debug starts ?

### ISSUE THREAD

**Issue by Marko021** [gwt-plugins/gwt-eclipse-plugin#338]
Hi Brendon,

We all know that GWT generators will go away so we need something instead. I know that Java annotation processors should be used instead of generators but the problem (this is only one of the multiple problems if you ask me) with processors is that they can "see" only one project they are registered in. I need something that can "inspect" multiple projects and to create one single factory class from multiple classes (that are in multiple projects/modules) with some particular annotation. I have created the class that is able to do that without annotation processors. I know that what I am asking can be achieved with maven, ant or some other build system but my question is : "Would it be possible to somehow hook something to be executed (generation of factories (Java source)) when user selects debug as "GWT ... (whatever option supported)" meaning: something (my jar/my class with Java main for example) will be executed before compiler starts building. This would be similar how generators are operating. Generator code (rebinding) is executed before debug process starts. Do you think something like this would be possible to achieve with the plugin? In another words, would it be possible to read something from some .ini file (or some other way like plugin settings window) that should be executed before you actually do the action you are executing when user clicks on Debug as GWT... ? 

Basically,  I am starting my class main method that is creating the factory before I start debugging (when I know that I have added something with my annotation) , my question is:  "Can I do this automatically before debug process starts (without maven goal but simply when I click debug as ... GWT ... in Eclipse)" ? 

Regards,
Marko

**Comment by Marko021** [comment]
There is no need to do this in the plugin. It can be done simply by specifying custom class main method in debug configuration instead of GWT. In custom main method first generate everything and than forward call to appropriate GWT main that starts application e.g. DevMode.main(args);

Closing this as it is not needed.

**Comment by branflake2267** [comment]
I was looking at Eclipse Oxygen Group Launcher, where you can run several launchers. This type of behaviors could be done using the group launcher.

**Comment by Marko021** [comment]
Yes I see now. There are also small plugin for composite launching of applications like this one: https://marketplace.eclipse.org/content/composite-launcher that works for all versions od Eclipse form Juno to Oxygen. Thanks for info. 

BTW, you got a star+ on the Marketplace from me a couple of minutes ago. Keep up the good work.

Cheers, 
Marko.

**Comment by branflake2267** [comment]
Nifty launcher. 

Awesome, thanks for the star! :) Thanks!
